### ISSUE TITLE
New False Negative for DM_STRING_CTOR

### ISSUE THREAD

**Issue by Belle-PL** [spotbugs/spotbugs#1767]
I found a false negative for this rule. I think we should consider this API because its internal implementation is to invoke new String(java.lang.String). This program can lead to lower performance.

The rule link: [link](https://spotbugs.readthedocs.io/en/latest/bugDescriptions.html#dm-method-invokes-inefficient-new-string-string-constructor-dm-string-ctor). Please refer to the following code sample:
```java
public void test(StringBuffer buf) {
    String str = new String(buf);  // Need report warning
}
```
My SpotBugs version is 4.4.2 and I think it needs report a warning for the line 2, but no expected outcome. So, this is a false negative. Thanks for your consideration.

**Comment by iloveeclipse** [comment]
As said, there is no plans to change anything, because that's not an issue.
